src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -272,6 +271,102 @@ export class CollapseCellsAction extends ToggleableAction {
 	}
 }
 
+export class RunParametersAction extends TooltipFromLabelAction {
+	private static readonly label = localize('runParameters', "Run with Parameters");
+	private static readonly iconClass = 'icon-run-with-parameters';
+
+	constructor(id: string,
+		toggleTooltip: boolean,
+		context: URI,
+		@IQuickInputService private quickInputService: IQuickInputService,
+		@INotebookService private _notebookService: INotebookService,
+		@INotificationService private notificationService: INotificationService,
+	) {
+		super(id, {
+			label: RunParametersAction.label,
+			baseClass: baseIconClass,
+			iconClass: RunParametersAction.iconClass,
+			maskedIconClass: maskedIconClass,
+			shouldToggleTooltip: toggleTooltip
+		});
+	}
+
+	/**
+	 * Gets Default Parameters in Notebook from Parameter Cell
+	 * Uses that as Placeholder values for user to inject new values for
+	 * Once user enters all values it will open the new parameterized notebook
+	 * with injected parameters value from the QuickInput
+	*/
+	public async run(context: URI): Promise<void> {
+		const editor = this._notebookService.findNotebookEditor(context);
+		// Set defaultParameters to the parameter values in parameter cell
+		let defaultParameters = new Map<string, string>();
+		editor.cells.forEach(cell => {
+			if (cell.isParameter) {
+				for (let parameter of cell.source) {
+					let param = parameter.split('=', 2);
+					defaultParameters.set(param[0].trim(), param[1].trim());
+				}
+			}
+		});
+
+		// Store new parameters values the user inputs
+		let inputParameters = new Map<string, string>();
+		let uriParams = new URLSearchParams();
+		// Store new parameter values to map based off defaultParameters
+		if (defaultParameters.size === 0) {
+			// If there is no parameter cell indicate to user to create one
+			this.notificationService.notify({
+				severity: Severity.Info,
+				message: localize('noParametersCell', "This notebook cannot run with parameters until a parameter cell is added. [Learn more](https://docs.microsoft.com/sql/azure-data-studio/notebooks/notebooks-parameterization)."),
+			});
+			return;
+		} else {
+			for (let key of defaultParameters.keys()) {
+				let newParameterValue = await this.quickInputService.input({ prompt: key, value: defaultParameters.get(key) });
+				// If user cancels or escapes then it stops the action entirely
+				if (newParameterValue === undefined) {
+					return;
+				}
+				inputParameters.set(key, newParameterValue);
```

Comment:
So right now it'd be required that the users include the string delimiters for string values as well - otherwise it'll break. Did you discuss this with design at all? I'm not sure that's going to be clear for people and so might cause some annoyances.

You could probably just follow up with this if you get some complaints - but I'd run it by some other people to get opinions on what to do here. It's tricky because you don't know for sure whether the param is supposed to be a string and so just blindly adding delimiters doesn't seem right either (even if it was a string originally maybe they want to enter a different param type now)

Reply:
I did run it by design, Raden and I discussed this. Users would have to include string delimiters for string values as we do not plan to do the validation at this moment. 

That is why I changed the setting from placeHolder to value in quickInput so that the user will know that for said parameter value it expects strings, as shown by the first parameter in the new gif I posted. We plan on addressing the concern however later as this ties in directly to the parameters validation aspect. For now, this will work.